pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -111,37 +174,30 @@ controlPlane:
     bindPort: 6443
 EOF
 
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-ca -ojson | jq '.data."tls.crt"' -r | base64 --decode > /etc/kubernetes/pki/ca.crt
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-ca -ojson | jq '.data."tls.key"' -r | base64 --decode > /etc/kubernetes/pki/ca.key
-
-mkdir -p /etc/kubernetes/pki/etcd
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-etcd-ca -ojson | jq '.data."tls.crt"' -r | base64 --decode > /etc/kubernetes/pki/etcd/ca.crt
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-etcd-ca -ojson | jq '.data."tls.key"' -r | base64 --decode > /etc/kubernetes/pki/etcd/ca.key
-
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-front-proxy -ojson | jq '.data."tls.crt"' -r | base64 --decode > /etc/kubernetes/pki/front-proxy-ca.crt
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system \
-get secret kubeadm-certs-front-proxy -ojson | jq '.data."tls.key"' -r | base64 --decode > /etc/kubernetes/pki/front-proxy-ca.key
-
-
-kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system get secrets kubeadm-sa-certs -ojson | jq '.data."sa-certs.tar.gz"' -r | base64 --decode > /etc/kubernetes/pki/sa-certs.tar.gz
-cd / 
-tar -xvf /etc/kubernetes/pki/sa-certs.tar.gz
-
 kubeadm join --config /tmp/kubeadm-controlplane-join-config.yaml --v 10
 `
 )
 
+func isKeyPairValid(cert, key string) bool {
```

Comment:
This should be renamed to something like "hasCertAndKey" and then the logic looks like it can be reduced to `return cert != "" && key != ""`